datadog/tests/provider_test.go

```diff
@@ -515,11 +516,11 @@ func TestProvider_impl(t *testing.T) {
 	var _ terraform.ResourceProvider = datadog.Provider()
 }
 
-func testAccPreCheck(t *testing.T) {
+func testAccPreCheck(ctx context.Context, t *testing.T) {
 	if isReplaying() {
 		return
 	}
-	_, finish := testSpan(context.Background(), t)
+	_, finish := testSpan(ctx, t)
 	t.Cleanup(finish)
```

Comment:
it doesn't seem like the right place to create test span as the context can not be passed to clients.